lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -707,7 +707,8 @@ pub trait SegmentOptimizer {
 
             timer.set_success(true);
 
-            if let Some(new_segment) = segments_lock.get(new_segment_id) {
+
+            if let Some(new_segment) = segments.read().get(new_segment_id) {
                 return Ok(new_segment.get().read().available_point_count());
             }
```

Comment:
We don't have to do this. We can count the numbers in `optimized_segment` before calling `swap_new`. That way we don't have to keep track of the new segment ID and don't have to do locking.

Reply:
Right!